enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,460 @@
+br-ext SDN switch acts as a transparent gateway and routes traffic towards
+Windows nodes.
+
+In the SDN live migration use case, we can enhance this feature to connect the
+nodes managed by different CNI plugins. To minimize the implementation effort
+and maintainability of the code, we will try to reuse the hybrid overlay code
+and only make necessary changes to both CNI plugins. OVN Kubernetes nodes
+(OVN-IC zones) will have full-mesh connections to all OpenShift SDN Nodes
+through VXLAN tunnel.
+
+On the OVN Kubernetes side, all the cross-CNI traffic shall follow the same path as the current hybrid overlay implementation. For OVN Kubernetes, we need to make the following enhancements:
+
+1. We need to prevent OVN Kubernetes from allocating a subnet for each host. We
+   need to reuse the host subnet allocated by OpenShift SDN.
+2. We need to modify OVN Kubernetes to allow overlapping between the cluster
+   network and the Hybrid overlay CIDR.
+3. OVN Kubernetes shall be able to handle adding/removing hybrid overlay nodes
+   on the fly.
+4. We need to allow `hybrid-overlay-node` to run on the Linux nodes using
+   OpenShift SDN as the CNI plugin; currently, it is designed to only run on
+   Windows nodes. It is responsible for:
+   - collecting the MAC address of the host primary interface and setting the
+    node annotation `k8s.ovn.org/hybrid-overlay-distributed-router-gateway-mac`.
+   - removing the pod annotations added by OVN Kubernetes for pods running on
+    the local node.
+
+On the OpenShift SDN side, when a node is converted to OVN Kubernetes, it will
+be almost transparent to the control-plane of OpenShift SDN. But we still need
+to introduce a 'migration mode' for OpenShift SDN by:
+
+1. Change ingress NetworkPolicy processing to be based entirely on pod IPs
+   rather than using namespace VNIDs, since packets from OVN nodes will have the
+   VNID 0 set.
+2. To be compatible with Windows node VXLAN implementation, OVN Kubernetes
+   hybrid overlay uses the peer host interface MAC as the VXLAN inner dest MAC
+   for egress traffic. Therefore, when packets arrive at the br0 of the SDN
+   node, they cannot be forwarded to the pod interface correctly. We need to add
+   a flow to change the dst MAC to the pod interface MAC for such ingress
+   traffic.
+
+### MTU Consideration
+
+There are two related but distinct MTU values to consider: the hardware MTU and
+the cluster MTU. The cluster MTU is the MTU value for pod interfaces. It is
+always less than your hardware MTU to account for the cluster network overlay
+overhead. The overhead is 100 bytes for OVN Kubernetes and 50 bytes for SDN. For
+example, if the hardware MTU is 1500, then the auto-probed cluster MTU would be
+1400 for OVN Kubernetes and 1450 for OpenShift SDN. During the migration, this
+MTU mismatch would break some cross-CNI traffic.
+
+To avoid this issue, before starting the migration, users need to decrease the
```

Comment:
Is MTU migration automated or not?
This sentence suggests that it is not but there is also this one:
```
CNO will check the current hardware MTU and the cluster MTU to see if there is enough free space for OVN-K overlay overhead
```

Reply:
MTU migration will be part of the live migration, so it will be automated. Let me make the sentences more clear.

Reply:
Just a clarification: it is only automated in the scope of this procedure because it doesn't contemplate the need to update the physical network MTU. This probably worth adding as well.